src/library/itunes/itunesfeature.cpp

```diff
@@ -370,12 +374,27 @@ TreeItem* ITunesFeature::importLibrary() {
                     if (playlist_root != NULL)
                         delete playlist_root;
                     playlist_root = parsePlaylists(xml);
+                    isTracksParsed = true;
                 }
             }
         }
     }
 
     itunes_file.close();
+    
+    if (isMusicFolderLocatedAfterTracks) {
+      qDebug() << "Updating iTunes real path from " << m_dbItunesRoot << " to " << m_mixxxItunesRoot;
+      // In some iTunes files "Music Folder" XML node is located at the end of file. So, we need to 
+      QSqlQuery query(m_database);
+      query.prepare("UPDATE itunes_library SET location = replace( location, :itunes_path, :mixxx_path )");
```

Comment:
1. ltrim() [search and remove specified characters](https://www.techonthenet.com/sqlite/functions/ltrim.php), not substring. So, only replace() can be used if we want to make it in one query. In other case we need to iterate over each row.
2. QSting.replace() also called in original replacement:
````
if (!m_dbItunesRoot.isEmpty()) {
    location.replace(m_dbItunesRoot, m_mixxxItunesRoot);
}